R/zchunk_L2242.land_input_4_irr_mgmt.R

```diff
@@ -1,90 +1,120 @@
 #'
 #' @param command API command to execute
 #' @param ... other optional parameters, depending on command
 #' @return Depends on \code{command}: either a vector of required inputs,
 #' a vector of output names, or (if \code{command} is "MAKE") all
-#' the generated outputs: \code{L2241.LN4_LogitTables[[ curr_table_name ]]}, \code{L2242.LN4_Logit}, \code{L2242.LN4_NodeGhostShare}, \code{L2242.LN4_NodeIsGhostShareRel}. The corresponding file in the
+#' the generated outputs: \code{L2242.LN4_Logit}, \code{L2242.LN4_NodeGhostShare}, \code{L2242.LN4_NodeIsGhostShareRel}. The corresponding file in the
 #' original data system was \code{L2242.land_input_4_irr_mgmt.R} (aglu level2).
-#' @details Describe in detail what this chunk does.
+#' @details This chunk generates the logit exponent of the fourth land nest by region, the ghost node share for bionenergy node in future years,
+#' and specifies whether the bionenergy ghost node share is relative.
 #' @importFrom assertthat assert_that
 #' @importFrom dplyr filter mutate select
 #' @importFrom tidyr gather spread
-#' @author YourInitials CurrentMonthName 2017
-#' @export
-module_aglu_L2242.land_input_4_irr_mgmt_DISABLED <- function(command, ...) {
+#' @author RC August 2017
+module_aglu_L2242.land_input_4_irr_mgmt <- function(command, ...) {
   if(command == driver.DECLARE_INPUTS) {
-    return(c( "L2241.LN4_Logit",
-              "L2241.LN4_NodeGhostShare",
-              "L2241.LN4_NodeIsGhostShareRel"))
+    return(c(FILE = "aglu/A_LandNode_logit_irr",
+             FILE = "temp-data-inject/L223.LN3_MgdAllocation_bio",
+             FILE = "temp-data-inject/L223.LN3_MgdAllocation_crop",
+             FILE = "temp-data-inject/L223.LN3_LeafGhostShare",
+             FILE = "temp-data-inject/L223.LN3_LeafIsGhostShareRel"))
   } else if(command == driver.DECLARE_OUTPUTS) {
     return(c("L2242.LN4_Logit",
              "L2242.LN4_NodeGhostShare",
              "L2242.LN4_NodeIsGhostShareRel"))
   } else if(command == driver.MAKE) {
 
+    # silence package check notes
+   GLU_name <- LandLeaf <- LandNode4 <- NULL
+
     all_data <- list(...)[[1]]
 
     # Load required inputs
-    L2241.LN4_Logit <- get_data(all_data, "L2241.LN4_Logit")
-    L2241.LN4_NodeGhostShare <- get_data(all_data, "L2241.LN4_NodeGhostShare")
-    L2241.LN4_NodeIsGhostShareRel <- get_data(all_data, "L2241.LN4_NodeIsGhostShareRel")
+    A_LandNode_logit_irr <- get_data(all_data, "aglu/A_LandNode_logit_irr")
+    L223.LN3_MgdAllocation_bio <- get_data(all_data, "temp-data-inject/L223.LN3_MgdAllocation_bio")
+    L223.LN3_MgdAllocation_crop <- get_data(all_data, "temp-data-inject/L223.LN3_MgdAllocation_crop")
+    L223.LN3_LeafGhostShare <- get_data(all_data, "temp-data-inject/L223.LN3_LeafGhostShare")
+    L223.LN3_LeafIsGhostShareRel <- get_data(all_data, "temp-data-inject/L223.LN3_LeafIsGhostShareRel")
+
+    names_LN3_Leaf <- c("region", "LandAllocatorRoot", "LandNode1", "LandNode2", "LandNode3", "LandLeaf")
```

Comment:
Not using `LEVEL2_DATA_NAMES`?

Reply:
`LN3_Leaf` currently is not in the `LEVEL2_DATA_NAMES` function. I tried adding it to the function file myself, but it seems not doing anything after rebuilding `gcamdata` and running the driver.

Reply:
Another good question for @pralitp

Reply:
Did you source the data file after update: `source("data-raw/level2_data_names.R")`

The call to `devtools::use_data(LEVEL2_DATA_NAMES, overwrite = TRUE)` at the bottom is important.  We set it up this way to follow the guidelines for handling data in R packages here: http://r-pkgs.had.co.nz/data.html

Reply:
Got it. Thanks @pralitp

Reply:
@pralitp If you get a chance please add this info to the wiki.

Reply:
I'm on it

Reply:
@bpbond, oops.  It looks like I already did but it is part of the XML conversion page.  Should it go else where?